File: converted/source2/source/zombie/asset/Asset$ObserverCallback.java
package zombie.asset;

import java.util.ArrayList;
import zombie.asset.Asset.State;

public final class Asset$ObserverCallback extends ArrayList<AssetStateObserver> {
   public void invoke(State var1, State var2, Asset var3) {
      int var4 = this.size();

      for (int var5 = 0; var5 < var4; var5++) {
         this.get(var5).onStateChanged(var1, var2, var3);
      }
   }
}

File: ProjectZomboidSwitch/src/zombie/asset/Asset$ObserverCallback.cpp

#include "zombie/asset/Asset$ObserverCallback.h"
#include "zombie/asset/AssetStateObserver.h"

namespace zombie::asset {

void Asset$ObserverCallback::invoke(State oldState, State newState, Asset& asset) {
    for (const auto& observer : m_observers) {
        if (observer) {
            observer->onStateChanged(oldState, newState, asset);
        }
    }
}

} // namespace zombie::asset
